libs/Helpers.py

```diff
@@ -160,3 +160,42 @@ def uninstall_managed_starburst_using_addon_flow(self, cluster_name):
         ocm_client = OpenshiftClusterManager()
         ocm_client.cluster_name = cluster_name
         ocm_client.uninstall_managed_starburst_addon(exit_on_failure=False)
+
+    @keyword
+    def inference_comparison(self, expected, received, threshold=0.00001):
+        try:
+            import numbers, decimal, ast
+
+            # Cast from string to python type
+            expected=ast.literal_eval(expected)
+            received=ast.literal_eval(received)
+
+            failures=[]
+            
+            def _inference_object_comparison(expected, received, threshold):
+                if isinstance(expected, dict):
+                    # if current element is a dict, compare the keys and then each value
+                    if not expected.keys()==received.keys(): failures.append([expected.keys(),received.keys()])
+                    for k in expected.keys():
+                        _inference_object_comparison(expected[k], received[k], threshold)
+                elif isinstance(expected, list):
+                    # if current element is a list, compare each value 1 by 1
```

Comment:
you're using dictionary right? do you think this function will be used for list too?

Reply:
even though it starts from a dictionary, it then compares element by element, and the dictionary might contain lists

Reply:
oh I missed the line where you call the same function in the inside for loop. Okay